src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -93,15 +93,16 @@ void GcsResourceManager::HandleGetAllAvailableResources(
       if (using_resource_reports) {
         auto resource_iter =
             node_resource_usages_[node_id].resources_available().find(resource_name);
-        if (resource_iter != node_resource_usages_[node_id].resources_available().end()) {
+        if (resource_iter != node_resource_usages_[node_id].resources_available().end() &&
+            resource_iter->second > 0) {
```

Comment:
If I look at the else statement, it seems like we update resources when the value > 0. Is it expected?

Reply:
Every resource in `node_resources.available` has value greater than 0, because `ResourceRequest` automatically erases any resource with zero value. So while iterating over `node_resources.available`, if `using_resource_reports` == `true`, we have to make sure no resource with zero value inserted. I believe there is a test requiring this behavior, but can not remember which one for now.

Reply:
If this behavior is important and we don't expect to report any 0 values, should we assert this instead?